hyperspy/_signals/sed.py

```diff
@@ -0,0 +1,465 @@
+                par_str += 'Please set them in the following wizard'
+                is_ok = messagesui.information(par_str)
+                if is_ok:
+                    self._set_microscope_parameters()
+                else:
+                    return True
+            else:
+                return True
+        else:
+            return False
+
+    def get_direct_beam_mask(self, radius=None, center=None):
+        """Generate a signal mask for the direct beam.
+
+        Parameters
+        ----------
+        radius : int
+            User specified radius for the circular mask.
+        center : tuple, None
+            User specified (x, y) position of the diffraction pattern center.
+            i.e. the direct beam position. If None it is assumed that the direct
+            beam is at the center of the diffraction pattern.
+
+        Return
+        ------
+        mask : array
+            The mask of the direct beam
+        """
+        r = radius
+        ny = self.axes_manager.signal_shape[1]
+        nx = self.axes_manager.signal_shape[0]
+
+        if center == None:
+            a = (self.axes_manager.signal_shape[1] - 1) / 2
+            b = (self.axes_manager.signal_shape[0] - 1) / 2
+        else:
+            a, b = center
+
+        y, x = np.ogrid[-b:ny-b, -a:nx-a]
+        mask = x*x + y*y <= r*r
+        return mask
+
+    def get_direct_beam_position(self, radius=None):
+        """Estimate the position of the direct beam in each SED pattern.
+
+        Parameters
+        ----------
+        radius : int
+            Defines the size of the circular region within which the direct beam
+            position is refined.
+
+        Return
+        ------
+        center: array
+            Estimated position (x, y) of the direct beam.
+
+        Notes
+        -----
+        Based on work presented by Thomas White in his PhD, Cambridge, (2009)
+        which itself built on Zaefferer (2000).
+        """
+        # sum images to produce image in which direct beam reinforced and take
+        # the position of maximum intensity as the initial estimate of center.
+        dp_sum = self.sum()
+        maxes = np.asarray(np.where(dp_sum.data == dp_sum.data.max()))
+        mref = [np.average(maxes[0]), np.average(maxes[1])]
+        # specify array of dims (nav_size, 2) in which to store centers and find
+        # the center of each pattern by determining the direct beam position.
+        arr_shape = (self.axes_manager.navigation_size, 2)
+        c = np.zeros(arr_shape, dtype=int)
+        for z, index in zip(self._iterate_signal(),
+                            np.arange(0, self.axes_manager.navigation_size, 1)):
+            # initialise problem with initial center estimate
+            c_int = z[mref[0], mref[1]]
+            mask = self.get_direct_beam_mask(radius=radius, center=mref)
+            z_tmp = z * mask
+            # refine center position with shifting ROI
+            if c_int == z_tmp.max():
```

Comment:
I think that I'm not following: doesn't `cint` contains the estimate of the centre? If yes, what's the point of checking it is equal to the maximum of z in the mask?